src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1

```diff
@@ -1958,62 +1958,82 @@ class Build: ADOSVTBase
-                            
+                        else {
+                            $this.CheckBuildLevelPRTriggerSecretSettings($pullRequestTrigger, $controlResult)
                         }
                     }
-                    else
-                    {
-                        $controlResult.AddMessage([VerificationResult]::Passed, "Pipeline secrets are not  marked as available to pull request validations of public repo forks.");
-                    }
                 }
-                else
-                {
-                    $controlResult.AddMessage([VerificationResult]::Passed, "Pipeline secrets are not marked as available to pull request validations of public repo forks.");
+                else {
+                    $this.CheckBuildLevelPRTriggerSecretSettings($pullRequestTrigger, $controlResult)
                 }
             }
-            else
-            {
-                $controlResult.AddMessage([VerificationResult]::Passed,"Pull request validation trigger is not set for build pipeline.");
+            else {
+                $controlResult.AddMessage([VerificationResult]::Verify, "Pull request validation trigger is not enabled for build pipeline.");
             }
         }
-        else
-        {
-            $controlResult.AddMessage([VerificationResult]::Passed,"No trigger is enabled for build pipeline.");
+        else {
+            $controlResult.AddMessage([VerificationResult]::Verify, "No trigger is enabled for build pipeline.");
         }
-
         return  $controlResult
     }
 
+    hidden [ControlResult] CheckBuildLevelPRTriggerSecretSettings([PSObject] $pullRequestTrigger,[ControlResult] $controlResult){
+        # initlizing $isRepoPrivate = $true as visibility setting is not available for ADO repositories.
+        $isRepoPrivate = $true
+        if ([Helpers]::CheckMember($this.BuildObj[0], "repository.properties.IsPrivate")) {
+            $isRepoPrivate = $this.BuildObj[0].repository.properties.IsPrivate
+        }
+        if (($isRepoPrivate -eq $false) -and [Helpers]::CheckMember($pullRequestTrigger, "forks")) {
+            if (($pullRequestTrigger.forks.enabled -eq $true) -and ($pullRequestTrigger.forks.allowSecrets -eq $true)) {
+                $controlResult.AddMessage([VerificationResult]::Failed, "Pipeline secrets are marked as available to pull request validations of public repo forks.");
+                if ($this.ControlFixBackupRequired -or $this.BaselineConfigurationRequired) {
+                    #Data object that will be required to fix the control
+                    $controlResult.BackupControlState = $pullRequestTrigger.forks.allowSecrets;
+                }
+                if ($this.BaselineConfigurationRequired) {
+                    $controlResult.AddMessage([Constants]::BaselineConfigurationMsg -f $this.ResourceContext.ResourceName);
+                    $this.CheckForkedBuildTriggerAutomatedFix($controlResult);           
```

Comment:
Please validate the automated fix for this control, as we have updated the code and it won't be affected